src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -57,224 +58,229 @@ pub fn is_min_const_fn(tcx: TyCtxt<'tcx>, def_id: DefId, body: &'a Body<'tcx>) -
-            check_operand(tcx, lhs, span, def_id, body)?;
-            check_operand(tcx, rhs, span, def_id, body)?;
-            let ty = lhs.ty(body, tcx);
-            if ty.is_integral() || ty.is_bool() || ty.is_char() {
-                Ok(())
-            } else {
-                Err((
-                    span,
-                    "only int, `bool` and `char` operations are stable in const fn".into(),
-                ))
+    fn visit_rvalue(
+        &mut self,
+        rvalue: &Rvalue<'tcx>,
+        location: Location,
+    ) -> McfResult {
+        let IsMinConstFn { tcx, span, body, .. } = *self;
+        match rvalue {
+            Rvalue::Cast(CastKind::Misc, operand, cast_ty) => {
+                use rustc::ty::cast::CastTy;
+                let cast_in = CastTy::from_ty(operand.ty(body, tcx))
+                    .expect("bad input type for cast");
+                let cast_out = CastTy::from_ty(cast_ty)
+                    .expect("bad output type for cast");
+                match (cast_in, cast_out) {
+                    | (CastTy::Ptr(_), CastTy::Int(_))
+                    | (CastTy::FnPtr, CastTy::Int(_))
+                    => return Err((
+                        span,
+                        "casting pointers to ints is unstable in const fn".into(),
+                    )),
+                    (CastTy::RPtr(_), CastTy::Float) => bug!(),
+                    (CastTy::RPtr(_), CastTy::Int(_)) => bug!(),
+                    (CastTy::Ptr(_), CastTy::RPtr(_)) => bug!(),
+                    _ => {}
+                }
             }
-        }
-        Rvalue::NullaryOp(NullOp::SizeOf, _) => Ok(()),
-        Rvalue::NullaryOp(NullOp::Box, _) => Err((
-            span,
-            "heap allocations are not allowed in const fn".into(),
-        )),
-        Rvalue::UnaryOp(_, operand) => {
-            let ty = operand.ty(body, tcx);
-            if ty.is_integral() || ty.is_bool() {
-                check_operand(tcx, operand, span, def_id, body)
-            } else {
-                Err((
-                    span,
-                    "only int and `bool` operations are stable in const fn".into(),
-                ))
+            | Rvalue::Cast(CastKind::Pointer(PointerCast::UnsafeFnPointer), _, _)
+            | Rvalue::Cast(CastKind::Pointer(PointerCast::ClosureFnPointer(_)), _, _)
+            | Rvalue::Cast(CastKind::Pointer(PointerCast::ReifyFnPointer), _, _)
+            => return Err((
+                span,
+                "function pointer casts are not allowed in const fn".into(),
+            )),
+            Rvalue::Cast(CastKind::Pointer(PointerCast::Unsize), _, _) => return Err((
+                span,
+                "unsizing casts are not allowed in const fn".into(),
+            )),
+            // binops are fine on integers
+            Rvalue::BinaryOp(_, lhs, _) | Rvalue::CheckedBinaryOp(_, lhs, _) => {
+                // FIXME: do we need to type check `rhs`?
```

Comment:
I don't think there are any binops where the rhs differs from the lhs and where the rhs is anything but an integer. So to answer the question: no